### ISSUE TITLE
2.0.7 Beta Test Reports

### ISSUE THREAD

**Issue by kiki-ja** [cleidigh/printing-tools-ng#80]
**All**
2.0.7 See the comments below for the development status of bate.
"v2.0.7 - Thunderbird v91 compatibility - Release Status" https://github.com/cleidigh/printing-tools-ng/issues/81#issue-976570906

@cleidigh

This is a report of PrintingTools NG 2.0.7b1.
OS: Windows 10 Pro 64bit 21H1
Thunderbird: 91.0.2 64bit ja version

1.Can be installed on Thunderbird 91.0.2.
2.PrintingTools NG option settings are not enabled.
3.The print dialog (preview) of Thunderbird itself opens and printing is possible.

Please see the screenshot.

**Print Preview [ja locale]**
![Print Preview Default  ja locale](https://user-images.githubusercontent.com/80428201/130360266-8b0b5c34-b2b5-4623-ab47-4a7b1a4f0a7f.png)

**Print Preview [en-GB locale]**
![Print Preview 2 0 7b1  en-GB locale](https://user-images.githubusercontent.com/80428201/130360301-b2b11f9b-9c5e-4713-a12f-ece0032eeca8.png)

**Comment by cleidigh** [comment]
@kiki-ja 
I should have posted b1 was just a skeleton beginning
I am hoping b2 May be somewhat functional
I will make a posting
I am doing a lot of investigation of the printing engine in 91
Christopher

**Comment by kiki-ja** [comment]
@cleidigh 

Thank you for the development status announcement of 2.0.7. #81
I will report the results of my testing.
I will also review the Japanese localization at the same time.

**Comment by kiki-ja** [comment]
@cleidigh 

Report of PrintingTools NG 2.0.7b1 & b2.

1. Width of Use fonts choose is narrow.
    [PTG Option - Style] 2.0.7b1 & b2
2. Width of Use fonts choose is narrow.
    [PTG Option - Option for addressbook printing] 2.0.7b2
3. The menu 'PTset;' will appear.
     [Thunderbird Menu - File] 2.0.7b2

Please see the screenshot.

**Option - Style - Use Fonts 2.0.7b1 & b2**
![2 0 7b2 - Option - Style - Font02](https://user-images.githubusercontent.com/80428201/130624985-97bc6bb7-a89b-4157-be6c-685a78970f89.png)

**Option - Option for addressbook printing - Use font 2.0.7b2**
![2 0 7b2- Option-Option for addressbook printing-Use font00](https://user-images.githubusercontent.com/80428201/130625167-edf8f4ee-7775-4a2f-8961-7c1ffce4e2df.png)

**Thunderbird Menu - File 2.0.7b2**
![2 0 7b2-Menu-File-PTG00](https://user-images.githubusercontent.com/80428201/130625654-2ab32dcf-3555-4173-9c0c-9fbcbacf722c.png)

**Comment by kiki-ja** [comment]
@cleidigh 

1. Many of the custom header displays are now enabled.
2. 'received date' and 'attachments' were not displayed (it's out).
3. The date and time was left at default and localization (eg. Japanese) was disabled.

Please see the screenshot.

**Print Preview 2.0.7b3  en-GB locale**
![Print Preview 2 0 7b3  en-GB locale](https://user-images.githubusercontent.com/80428201/131254319-af20e309-d90d-4260-90f5-edf4a9a50fdb.png)

**Print Preview 2.0.7b3  ja locale**
![Print Preview 2 0 7b3  ja locale](https://user-images.githubusercontent.com/80428201/131254414-2276311f-7d91-46da-b32f-3394d4bba15a.png)

**Comment by kiki-ja** [comment]
@cleidigh 

Report of Thunderbird 78.13.0 64bit + PrintingTools NG 2.0.7b3.
OS: Windows 10 Pro 64bit 21H1

1. In Thunderbird 78.13.0, the Options panel of PrintingTools NG 2.0.7b3 opens normally.
2. However, the print preview stays at Thunderbird default.

I think PrintingTools 2.0.7 should be used for Thunderbird version 91.0 or later.
The reason for this is that the printing specifications have changed significantly from the versions up to 78.0.

Please see the screenshot.

**Print Preview PrintingTools NG 2.0.7b3 + 78.13.0 en-US locale**
![Print Preview 2 0 7b3 - 78 13 0  en-US](https://user-images.githubusercontent.com/80428201/131256075-0e51c9ce-00a0-439b-8d00-c46d746f3d77.png)

**Print Preview PrintingTools NG 2.0.7b3 + 78.13.0 ja locale**
![Print Preview 2 0 7b3 - 78 13 0  ja](https://user-images.githubusercontent.com/80428201/131256123-e0501ed8-fecf-47a4-b364-dca4198baa16.png)

**Comment by cleidigh** [comment]
@kiki-ja 
thanks for the testing
right now everything is waiting on the ability to refresh
the preview 
I am awaiting for help
I will update
Christtopher

**Comment by Alpengreis** [comment]
Here additionally to the known refresh problem I have the problem that I can't change the font and font size for the address book (reverts back to exist values).

**Comment by kiki-ja** [comment]
@cleidigh

This is a report on Printing Tools NG 2.0.7-b4.

OS: Windows 10 Pro 21H2 64bit
Thunderbird 91.5.1 64bit

1. Many, but not all, of the options worked in b3.
2. I don't think all of the options work in b4.
Currently, I think b3 is better.
*The header customization does not show attachments with icons.

**Print Preview / PrintingTools NG 2.0.7b4 + Thunderbird 91.5.1**
![Print Preview 2 0 7b4-0](https://user-images.githubusercontent.com/80428201/151003412-db8fbcfc-4468-42f7-acde-85143fcefdb2.png)

**Print Preview / PrintingTools NG 2.0.7b3 + Thunderbird 91.5.1**
![Print Preview 2 0 7b3-0](https://user-images.githubusercontent.com/80428201/151003474-61f270ba-2993-4cdf-8d76-474d2fb69a9f.png)

**Comment by kiki-ja** [comment]
@cleidigh

This is a report of Printing Tools NG 2.0.7-b6.
OS: Windows 10 Pro 21H2 64bit
Thunderbird 91.5.1 64bit

１．Print Preview in the menu bar [File] > [PTNG Print] now shows attachments name and icons in the header. good!
２．There is a difference in the display of the size of attached files.
３. [Print] with message selected now prints normally in Thunderbird.

**Print Preview / PrintingTools NG 2.0.7-b6 + Thunderbird 91.5.1 en-US**
![Print Preview 2 0 7-b6 en-US](https://user-images.githubusercontent.com/80428201/151691810-e06168b6-ba0b-4a44-ac9a-e75fe10fde4a.png)

**Print Preview / PrintingTools NG 2.0.7-b6 + Thunderbird 91.5.1 ja**
![Print Preview 2 0 7-b6](https://user-images.githubusercontent.com/80428201/151691814-7cd4e014-9540-4826-a3b1-c492d0f13a03.png)

**Question:**
Is it correct to understand that [PTNG Print] is a test print mode?

**Comment by querixg** [comment]
@cleidigh

Thunderbird 91.5.0 64bit
Linux Mint 20.1

Great to see the tool beeing made compatible with Thunderbird 91. Unfortunately the only two features I need don't work yet in 2.0.7-b6:

1. Printing selected text only
2. Cancelling the list of attachments at the end of the printout (while listing them in the header, which works fine already)

Moreover, the tool doesn't work if the mail is opened in a new window. I still quarrel with tabs.

**Comment by kiki-ja** [comment]
@querixg
@cleidigh

>1.Printing selected text only

It's the same for us.

>2.Cancelling the list of attachments at the end of the printout (while listing them in the header, which works fine already)

The ability to cancel attachments at the end of printing is not available in version 2.0.61.
Are you requesting additional features?
It can be shown or hidden by setting the option in Printing Tools NG or in Thunderbird beforehand.

>Moreover, the tool doesn't work if the mail is opened in a new window. I still quarrel with tabs.

It is the same for us. It is the same in version 2.0.61.
Are you requesting additional features?

**[PTNG Option > Header order]**
![PTNG Header order](https://user-images.githubusercontent.com/80428201/151775993-1eba2d23-180c-4e86-99b6-bc85e77c0ba5.png)

**[Menu bar > View / Display Attchments Inline ON/OFF]**
![Display Attachments Inline](https://user-images.githubusercontent.com/80428201/151776031-706f6d72-1e83-4262-af2a-00ea949af869.png)

**[Hide Attachments list and inline display]**
![Hide Attachment List and Inline](https://user-images.githubusercontent.com/80428201/151776695-d435d204-21ff-4d76-b284-a927a5194edd.png)

**Comment by kiki-ja** [comment]
@cleidigh

Printing a selected range worked for me with version 2.0.7-b3.
Is this helpful?

**[Select Message]**
![Print select message](https://user-images.githubusercontent.com/80428201/151791795-06ce93d9-a02c-484f-8ccb-936afa2862cb.png)

**[Print preview]**
![Print select message review](https://user-images.githubusercontent.com/80428201/151791804-a895e5c5-c580-400e-b1ee-278c98d4ba52.png)

**Comment by madvinegar** [comment]
In my case, it does not work, unless I check from the thunderbird print options the box that says "format - selection", but in this case it only prints out the selected text without the headers (date, subject, from, to.. etc.).

**Comment by madvinegar** [comment]
Thanks for your reply.
First of all, in order to see the PTNG Print option, I need to set and open the emails in tabs and not in separate windows. Otherwise the PTNG print option is not there.

Secondly, where can I find the 2.0.7-b7 ? I only see 2.0.7-b6

**Comment by madvinegar** [comment]
I tried to print with PTNG however "print selection" does not work. I am using 2.0.7-b6 with TB 91.5.1.
See the picture. Even if I selected a part of the text, it still shows that the whole email will be printed. (it is a spam email I used as an example).
<img width="1440" alt="Στιγμιότυπο 2022-01-31, 8 51 31 μμ" src="https://user-images.githubusercontent.com/11718763/151855141-d60b60ba-5cd2-43e2-9a7f-086c02937ba8.png">

**Comment by kiki-ja** [comment]
@madvinegar 

>First of all, in order to see the PTNG Print option, I need to set and open the emails in tabs and not in separate windows. Otherwise the PTNG print option is not there.

The "PTNG Print" will only appear if the message is opened in the message pane or in a tab.
That is the current beta specification.

>Secondly, where can I find the 2.0.7-b7 ? I only see 2.0.7-b6

I'm sorry.
b7 is a mistake. The correct number is b6.

**Comment by kiki-ja** [comment]
@madvinegar 

Thunderbird 91.5.1 64bit + Printing Tools NG 2.0.7-b6
Windows 10 Pro 21H2 64bit

The "print selection" doesn't work for me as well. ("PTNG Print")
This is the current (2.0.7-b6) situation.

**Comment by madvinegar** [comment]
I really hope this can be fixed. It is the only thing I need from the printing tools addon. :)
It is the reason that in my work computer I am still at TB version 68. :)

**Comment by cleidigh** [comment]
All
Here is the status on  selection:

- this is  what I  am working on for b7
- i dont see issues yet
- some questions may arise on format changes

@querixg 
you mentioned multiple selections, i do not believe that this is possible,
i also do not think this was ever done
if tb does allow it i can look into it 
I am not able to really test that myself

**Comment by cleidigh** [comment]
@querixg 
Regarding  attachments, I  think I understand your comment , however, I am not able to reproduce what I believe you are seeing 
Can you post some pics?
@cleidigh

**Comment by kiki-ja** [comment]
@querixg

I'm sorry.
The "List of attachments" you are referring to is the one that appears below the message body.
It's the part with the red border in the attached screenshot.
This is the default display feature of Thunderbird. It was introduced in version 91.0.
If you don't want this display style="design", please tell me what exactly you want.

**[Default Attachment List Display]**
![Default Attachment List](https://user-images.githubusercontent.com/80428201/152164239-53d7dc42-5fdc-4d75-95cb-f1bbc9e0314f.png)

>Text selection: I tried 2.0.7-b3, but it didn't work at all and the add-on even disappeared when Thunderbird had been started anew.

Do you have the option "Print just selection, when it exists" checked in PrintingTools NG 2.0.7-b3, circled in red in the attached screenshot?

**[Print Text Selection]**
![Print Text Selection](https://user-images.githubusercontent.com/80428201/152164273-f3721638-c74c-4b16-9bbf-2063ded3e293.png)

If it's checked and still doesn't work, then it's probably another problem.
The Thunderbird profile may be corrupted.
Try installing and testing only PrintingTools NG 2.0.7-b3 in a new Thunderbird profile.

**Comment by querixg** [comment]
@cleidigh
@kiki-ja
Thank you very much for taking care of my problems!

1. Text selection:

'Print just selection, when it exists' is my default setting. Having installed 2.0.7-b3 instead of -b6, choosing 'Print...' behaves like Thunderbird without PTNG: choosing selection cancels the header, but with some additional curiosities, choosing 'Ptest;' has no effect at all. Now I could not reproduce the tool disappearing on restart as mentioned earlier. 

2. List of attachments:

In Thunderbird's native printout there is this nasty listing of attachments at the end of the printout, the more attachments the more nasty. This has been so for years, it's not new in 91.0.

In the printout of  PTNG 2.0.61 (Thunderbird 78.14.0) and former versions of the tool it is missing, great! Having another computer not yet updated I can show this, displaying or hiding attachments in the header doesn't matter:
![2 0 61 hidden](https://user-images.githubusercontent.com/87893156/152192853-3939ab28-9ad1-4f56-8280-b433ee0642a0.png)
![2 0 61 not hidden](https://user-images.githubusercontent.com/87893156/152192996-57ef73d2-90b8-45de-809c-358b33136cd3.png)

In the printout of  PTNG 2.0.7-b6 it is present, displaying or hiding attachments in the header again doesn't matter:
![2 0 7-b6 hidden](https://user-images.githubusercontent.com/87893156/152193118-c18422b4-7545-4312-a5fb-2fbd0199bc3e.png)
![2 07-b6 not hidden](https://user-images.githubusercontent.com/87893156/152193201-529ae649-e3b4-4b07-8bba-bac940659083.png)

'Display attachments inline' in the settings of View (not my default setting) adds the attachments themselves to the body, 'hide inline attachments' in the settings of PTNG doesn't change anything of the printout with 2.0.61 as well as with 2.0.7-b6.

**Comment by cleidigh** [comment]
@madvinegar 
@querixg 
you will be happy to know that selection print should be good in b7!

please test

**Comment by cleidigh** [comment]
@querixg 
I still am not able to reproduce your atts list at end issue 
we need an example email 
pics of your options and any other stuff to try and reproduce 

I checked de locale also 

we will figure it out 

@cleidigh

**Comment by cleidigh** [comment]
> @querixg
> 
> I'm sorry. The "List of attachments" you are referring to is the one that appears below the message body. It's the part with the red border in the attached screenshot. This is the default display feature of Thunderbird. It was introduced in version 91.0. If you don't want this display style="design", please tell me what exactly you want.
> 
> **[Default Attachment List Display]** ![Default Attachment List](https://user-images.githubusercontent.com/80428201/152164239-53d7dc42-5fdc-4d75-95cb-f1bbc9e0314f.png)
> 
> > Text selection: I tried 2.0.7-b3, but it didn't work at all and the add-on even disappeared when Thunderbird had been started anew.
> 
> Do you have the option "Print just selection, when it exists" checked in PrintingTools NG 2.0.7-b3, circled in red in the attached screenshot?
> 
> **[Print Text Selection]** ![Print Text Selection](https://user-images.githubusercontent.com/80428201/152164273-f3721638-c74c-4b16-9bbf-2063ded3e293.png)
> 
> If it's checked and still doesn't work, then it's probably another problem. The Thunderbird profile may be corrupted. Try installing and testing only PrintingTools NG 2.0.7-b3 in a new Thunderbird profile.


@kiki-ja is correct,selection is only working now with b7

**Comment by kiki-ja** [comment]
All.

Correction.

>This is the default display feature of Thunderbird. It was introduced in version 91.0.

My comment above was a mistake.
It was designed to be displayed in Thunderbird 78.0 series.

**Comment by kiki-ja** [comment]
@cleidigh

This is a report on PrintingTools NG 2.0.7-b7.

OS: Windows 10 Pro 21H2 64bit
Thunderbird 91.5.1 64bit

１．Select text and printing will work. Good!
２．The attachment size in the header is displayed more accurately. Good!
３．The option "Hide inline attachments" does not seem to be working.
Attachments are printed inline even when this is checked.
PTNG 2.0.61 did not seem to work either.

**Attention!**
When printing with PrintingTools NG 2.0.7-b7, you need to go to the Menu bar > [File] > [Print…]

**[Print Preview All Text PTNG 2.0.7-b7]**
![Print Preview 2 0 7-b7](https://user-images.githubusercontent.com/80428201/152302717-ddae6063-dcc2-41d6-a437-3d362a86ee17.png)

**[Print Text Selection PTNG 2.0.7-b7]**
![Print Text Selection 2 0 7-b7](https://user-images.githubusercontent.com/80428201/152302735-5e06a7f1-57c0-47dd-a7a4-5e92e8658fd8.png)

**[Hide inline attachments Option PTNG 2.0.7-b7]**
![Hide inline attachments](https://user-images.githubusercontent.com/80428201/152302757-17098c9a-47e5-4b51-bbea-ce5ac8f75730.png)

**Comment by kiki-ja** [comment]
@querixg 

Is the following information correct for your needs?

-----
Purpose:
I would like to print a summary of my message.
I would like to print out a summary of a message with only the parts I need.

１．I want to hide the attachments list that Thunderbird displays at the bottom.
I would like to add a toggle option to PrintingTools NG to hide it.

**[Hide display / Attachments list of bottom]**
![querixg-06](https://user-images.githubusercontent.com/80428201/152312881-2c8310b2-b7fd-4e3b-bc9d-da6ba56002f5.png)

２．I want to select multiple arbitrary texts in the message body and print them.
Please add this function to the "Print just selection, when it exists" option.

**[Selection texts print / Multiple texts]**
![querixg-09](https://user-images.githubusercontent.com/80428201/152312900-1e310d99-8db4-4c4b-bb0b-27ad9fc98a6f.png)

**Comment by madvinegar** [comment]
> @madvinegar @querixg you will be happy to know that selection print should be good in b7!
> 
> please test

Yes! It works great and as it should! 
Ofcourse in order to work you need to open the email in a tab (and not in a separate window). But still, that's a minor price to pay. Many thanks @cleidigh !!!

**Comment by querixg** [comment]
@kiki-ja
Your pictures show exactly what I wanted, thank you. Maybe I should learn to create something like that myself. I hope I don't have to provide any more information now.

@cleidigh
'Print just selection' works fine with 2.0.7-b7 (single selection, i.e. printing only the first part selected, of course). Many thanks!
Attachments: Can't you reproduce what you did in 2.0.61? That worked fine as well.

**Comment by cleidigh** [comment]
all

can anyone reproduce what @querixg sees with an attachment list at bottom?

i never see this 
@cleidigh

**Comment by cleidigh** [comment]
all
I can reproduce the issue of not printing from a window 
I should be able to fix

**Comment by cleidigh** [comment]
All

quick b8 notes 

- fixed window print, may have edge  conditions 
- removed attachments table, will make optional in b9
going to work on in line images next 
@cleidigh

**Comment by kiki-ja** [comment]
@cleidigh

This is a report on Printing Tools NG 2.0.7-b8.

OS: Windows 10 Pro 21H2 64bit
Thunderbird 78.1.4 64bit

It does not work properly with Thunderbird 78.1.4.
In manifest.json, "strict_min_version": "68.0" should be changed to "91.0".

@kiki-ja

**Comment by kiki-ja** [comment]
@cleidigh

>    ・fixed window print, may have edge conditions
>    ・removed attachments table, will make optional in b9
>     going to work on in line images next

It's very GOOD and FINE!

**[Print Preview 2.0.7-b8]**
![Print Preview 2 0 7-b8](https://user-images.githubusercontent.com/80428201/152467532-32eb2322-2118-4084-b3ef-fd9b88b7f052.png)

@kiki-ja

**Comment by kiki-ja** [comment]
**All.**

**Attention!**
When printing with PrintingTools NG 2.0.7-b8, you need to go to the Menu bar > [File] > [Print…]

@cleidigh is working on integrating it into the context menu of Thunderbird as well.
@cleidigh wrote:
>PTNG print is temporarily being used to do experiments
>until i figure out menu wireing

From:https://github.com/cleidigh/printing-tools-ng/issues/81#issuecomment-1026467955

@kiki-ja

**Comment by librelegal** [comment]
This is a report on Printing Tools NG 2.0.7-b8.

OS: Windows 10 Pro 21H2 64bit
Thunderbird 91.5.1 32bit
[Print go to the Menu bar > [File] > [Print…] 
The required frame is set, attachments are only in frame, and not at end, just as it should, perfect
When printing with a right klick on the message and then --> print it does not work
When printing the message in a new tab going over the pull down menu at the right, it does not work

**Comment by librelegal** [comment]
This is a report on Printing Tools NG 2.0.7-b8.
OS: debian 10
Thunderbird 91.5.0 (64-Bit)
[Print go to the Menu bar > [File] > [Print…]
The required frame is set, attachments are only in frame, and not at end, just as it should, perfect
When printing with a right klick on the message and then --> print it does not work
When printing the message in a new tab going over the pull down menu at the right, it does not work
it shows the same behaviour as under Wind10

**Comment by kiki-ja** [comment]
@cleidigh
All.

This is a report on Printing Tools NG 2.0.7-b8.
OS: Windows 10 Pro 21H2 64bit
Thunderbird 91.5.1 64bit

Hide the images option worked.

@kiki-ja

**Comment by cleidigh** [comment]
@librelegal 
check out b9 i enabled msg context print also the " hamburger" menu

@cleidigh

**Comment by cleidigh** [comment]
b9 notes : 

- should be functionaly complete  - v2.0.6 equivalent 
- remove inline attachment list option - localized 
- add msg context, app menu print 
- Hide AB support  - no ab in release 
- 
There is an important missing piece for a clean shutdown to restore 
the overriden print menus
The code is also a spaghetti mess that would embarrase anyone ,
cleanup will take a bit.
@cleidigh

**Comment by cleidigh** [comment]
All 

test request : 

can someone test b9 opening an eml file?
this is issue #5  and i think it should work ,i cannot test myself due to setup issues 

@cleidigh

**Comment by querixg** [comment]
@cleidigh
2.0.7-b9 (Thunderbird 91.5.0, Linux Mint 20.1): Opening an eml file, File > Print does not respond. Just as with mail Print from main toolbar prints as without PTNG.

**Comment by kiki-ja** [comment]
@cleidigh 

2.0.7-b9 test.
OS: Windows 10 Pro 21H2 64bit
Thunderbird 91.6.0 64bit

I got the same result as @querixg.

- Open eml file
- File > Print > Nothing happen
- Hamburger Menu button > Print > Normal Print (see screenshot)

**[Print Preview / open eml file / Hamburger Menu button > Print]**
![Print Preview eml File Hamburger Menu on Window](https://user-images.githubusercontent.com/80428201/152754572-b7ee182a-04bc-428e-a8b1-7171cce6650c.png)

@kiki-ja

**Comment by kiki-ja** [comment]
@cleidigh 

2.0.7-b9 test report.
OS: Windows 10 Pro 21H2 64bit
Thunderbird 91.6.0 64bit

This is the test result when the message is opened in a new window.
- File > Print > PTNG print
- Hamburger menu > Print > Normal print
- Context menu button > Print > Normal print

![Open Message in Window 2 0 7-b9](https://user-images.githubusercontent.com/80428201/152762751-bc13058e-5d0c-4a1b-b83d-d27c15e41442.png)

@kiki-ja

**Comment by RRoenn** [comment]
> 2.0.7-b9 test report. OS: Windows 10 Pro 21H2 64bit Thunderbird 91.6.0 64bit
> 
> This is the test result when the message is opened in a new window.
> 
>     * File > Print > PTNG print
> 
>     * Hamburger menu button > Print > Normal print
> 
>     * Context menu > Print > Normal print

@kiki-ja 
What happen if you press Ctrl+P?

**Comment by kiki-ja** [comment]
@RRoenn 

>> 2.0.7-b9 test report. OS: Windows 10 Pro 21H2 64bit Thunderbird 91.6.0 64bit
>>
>> This is the test result when the message is opened in a new window.
>>
>>     * File > Print > PTNG print
>>
>>     * Hamburger menu button > Print > Normal print
>>
>>     * Context menu > Print > Normal print
　
> @kiki-ja 
> What happen if you press Ctrl+P?

Yes, a mouse click on "Print" and a "Ctrl+P" press had the same result.

@kiki-ja

**Comment by cleidigh** [comment]
@RRoenn 
b9 does not have all the message window menus enabled 
b10 will have those enabled 
b9 is also messed up for options  -screw up

@cleidigh

**Comment by kiki-ja** [comment]
@cleidigh 

This is a report on Printing Tools NG 2.0.7-b10.

OS: Windows 10 Pro 21H2 64bit
Thunderbird 97.0b3 64bit

It worked with Thunderbird 97.0b3.
However, Thunderbird 98.0a Daily is unstable and only partially works.
In manifest.json, I suggest adding "strict_max_version" to set it to "97.0".

@kiki-ja

**Comment by kiki-ja** [comment]
@cleidigh 
All.

2.0.7-b10 test report.
OS: Windows 10 Pro 21H2 64bit
Thunderbird 91.6.0 64bit

When printing **HTML messages**, there are the following differences between Thunderbird itself and PTNG 2.0.7-b10.
- Font size is different
- Image layout is different
  
A screenshot is attached.
![Compara image 比較イメージ-03](https://user-images.githubusercontent.com/80428201/153450349-d0262952-f909-4202-a923-805410494695.png)

A PDF file printed with PrintingTools NG 2.0.61 is attached for  reference.
The eml file of the **HTML message** used for the test is also attached as a compressed file (zip).

[PTNG 2.0.61 - Truly international browsing is here.pdf](https://github.com/cleidigh/printing-tools-ng/files/8042588/PTNG.2.0.61.-.Truly.international.browsing.is.here.pdf)
[Normal - Truly international browsing is here.pdf](https://github.com/cleidigh/printing-tools-ng/files/8042590/Normal.-.Truly.international.browsing.is.here.pdf)
[test_sample_files.zip](https://github.com/cleidigh/printing-tools-ng/files/8042591/test_sample_files.zip)


I am not sure which is more correct for printing **HTML messages**, Thunderbird itself or PTNG 2.0.7-b10.
If you prioritize the release of version 2.0.7, you may want to put this issue on hold or pend.

@kiki-ja

**Comment by cleidigh** [comment]
@kiki-ja 
All

Our new toolbar button, comments ?

@cleidigh 
![image](https://user-images.githubusercontent.com/25272315/153523416-28799a92-09bb-4823-8a9f-8b757fa2edb3.png)

**Comment by cleidigh** [comment]
@kiki-ja 

I think you are seeing the same thing as @riener

I will try to investigate

**Comment by kiki-ja** [comment]
@cleidigh 
All
 
> Our new toolbar button, comments ?
 
Oh, that's wonderful!
I think it will be very easy to use.
Thank you so much.

@kiki-ja

**Comment by kiki-ja** [comment]
@cleidigh 
All

2.0.7-**b11** test report.
OS: Windows 10 Pro 21H2 64bit
Thunderbird 91.6.0 64bit

This is the result of the operation check of the File menu (Print... (NG)), App menu, Context menu, Shortcut menu (CTRL+P),  and Icon button menu.

### **[3 Pane view mode]** ###
- File -> Print... (NG) -> ***Nothing is happen***
- App menu -> Print... (NG) -> ***Nothing is happen***
- Context menu -> Print... (NG) -> ***Nothing is happen***
- Ctrl+P -> ***Nothing is happen***
- Header view toolbar print icon button -> TB Normal print.
- PTNG icon button in toolbar -> All woks.

### **[Open Message in New Tab]** ###
- File -> Print... (NG) -> ***Nothing is happen***
- App menu -> Print... (NG) -> ***Nothing is happen***
- Context menu -> Print... (NG) -> ***Nothing is happen***
- Ctrl+P -> ***Nothing is happen***
- Header view toolbar print icon button -> TB Normal print.
- PTNG icon button in toolbar -> All woks.

### **[Open Message in New Window]** ###
- File -> Print... (NG) -> ***Nothing is happen***
- App menu -> Print... (NG) -> ***Nothing is happen***
- Context menu -> Print... (NG) -> ***Nothing is happen***
- Ctrl+P -> ***Nothing is happen***
- Header view toolbar print icon button -> TB Normal print.
- PTNG icon button in toolbar -> ***Nothing is happen***

### **[PTNG options]** ###
- File - > Printing options -> Open.
- PTNG icon button in toolbar -> PrintingTools NG Options -> Open.

@kiki-ja

**Comment by reinerblock** [comment]
Hm, I do not have those problems. In neither of the different views.

**Comment by reinerblock** [comment]
Like I already wrote by email. I don't have any new toolbar button. :(

![grafik](https://user-images.githubusercontent.com/5745579/153766430-f3181263-2c19-4c7c-975f-aeb6ce641bac.png)

The printer icon in the middle is just the default print icon.

**Comment by cleidigh** [comment]
@kiki-ja 
all

I forgot to mention that you should uninstall b10, reboot then install b11
it sounds like you have conflicting stuff from b10

@tidebre67 - i realized i missed an event stop for the window  case, fix in b12, cont P not working?
@reinerblock 
all i also forgot to mention that I do not install the toolbar button automatically 
you customize by dragging from the toolbar palate 

@cleidigh

**Comment by cleidigh** [comment]
@tidebre67 
yes i added something and messed up other stuff 
i rush to post overnight, 
@cleidigh

**Comment by kiki-ja** [comment]
@cleidigh 
All

2.0.7-b12 test report.
OS: Windows 10 Pro 21H2 64bit
Thunderbird 91.6.1 64bit

App Menu, Context Menu, Print... (NG), Ctrl+P behavior regression has been fixed.

@kiki-ja

**Comment by kiki-ja** [comment]
@cleidigh 
All

2.0.7-b13 test report.
OS: Windows 10 Pro 21H2 64bit
Thunderbird 91.6.1 64bit

This section describes the results of executing the commands in each menu.

App Menu, Context Menu, Print... (NG), Ctrl+P, and PTNG icon button work when a new message is displayed.
However, some of the same messages will work on the second and subsequent iterations, while others will not.
This varies depending on the viewing mode of the message.
Only the "Printing options" work in all viewing modes.

@kiki-ja

**Comment by cleidigh** [comment]
@kiki-ja 
all
sorry I have been struggling a lot  with the internal  print mechanism 
i may have a solution 

the outstanding issues 👍 
html differences
consistent  prints

i want the print stability  first 
I will probably post b14 tomorrow 

@cleidigh

**Comment by kiki-ja** [comment]
@cleidigh 
All

>sorry I have been struggling a lot with the internal print mechanism
>i may have a solution

I merged the new background.js and printingtoolsng-pengine.js into b13 and that improved things.
In b14p1, "Print... (NG)", "Ctrl+P" did not work for messages with attachments. 
>i want the print stability first

I think so too.

@kiki-ja

**Comment by cleidigh** [comment]
i have further changes 
should help

i do not have any support for attached messages and i dont think pt ever did
i would like to look at that for a point release 

i will also try to capture where we are later 
I really think we need to release asap

@cleidigh

**Comment by kiki-ja** [comment]
@cleidigh 
All

Here are my suggestions at this point （2.0.7-b14).

Looking back on the development of the beta version so far, why don't we postpone the following issues and prioritize the release of the stable version?
(Stable version means a version that works reliably.)

- [ ] Selecting multiple texts and printing them #80
- [ ] Print eml files and attachmments to messages #5
- [ ] Font size and image layout are different in HTML message printing #95
- [ ] Integrate PTNG with FiltaQuila add-on for automatic printing #94
- [ ] Still problem with page break after image #96

The reasons for this are
1. Solving problems requires a lot of analysis and takes a lot of time.
2. developing new features takes a lot of time as well
3. Christopher Leidigh's physical and mental strain seems to be continuing.

@kiki-ja

**Comment by reinerblock** [comment]
In beta 13 there is still the “old” problem of a page break after an image.

I already sent the email as EML via email. :)

![grafik](https://user-images.githubusercontent.com/5745579/154935877-78d982b6-7bd0-4bb6-831a-ab1bba4dbf3f.png)

**Comment by cleidigh** [comment]
@reinerblock 

have not forgotten you!
i wanted to get debugging in place 
lets try looking at the output using the new debug options described above and discuss in #95 

First try passthrough to see engine without formatting is the same 
@cleidigh

**Comment by reinerblock** [comment]
With beat 14 I came acros something strange.

First I changed the default printer to PDF-XChange Standard and added "initialsource finaloutput" for debugging.

In the preview of the email I used before suddenly my jaw hung wide open and I changed the default printer back to my Brother MFC-J480DW and opened the print preview again and got the following result.

![grafik](https://user-images.githubusercontent.com/5745579/155604422-b70f921c-ae1f-4b9c-b34c-dd0cb43af755.png)

No surprise here, or?
But then I changed the default printer again to PDF-XChange but without the debug parameters and the result was the following.

![grafik](https://user-images.githubusercontent.com/5745579/155604641-c988db91-57ef-4920-8f2a-a56261fd6ff0.png)

As you can see the text beginns in this case as it should directly below the image.
**BUT** the header is now wrong.

Now you see me confused. ;)

**Comment by cleidigh** [comment]
@reinerblock 
let's track this in #95 
i am going to close this soon so it's easier to manage 
in your second shot the header is partly correct meaning ptng was invoked 
did you look at debug?

**Comment by reinerblock** [comment]
No, but I'll do it now and send the result as soon as possible. But in about 10 min. today's lesson of Change Management starts and I've to squeeze it in somewhere. ;)

**Comment by reinerblock** [comment]
O.k. now I'd like to now, where to find the debug output.
I didn't find any settings that could cause the lack of the background color. :(
But there are a lot of settings in PDF-XChange.

![grafik](https://user-images.githubusercontent.com/5745579/155669866-6f685435-7da4-4598-af6d-a66bd96f7d4f.png)

**Comment by reinerblock** [comment]
Found the culprit.

![grafik](https://user-images.githubusercontent.com/5745579/155672388-32c59aa8-a340-432a-b7d9-1da2bc0773b3.png)

The PDF-XChange driver needs the "Print backgrounds" setting manually activated while it is automatically activated when printing with my Brother MFC-J480DW.

**Comment by querixg** [comment]
2.0.7-b14 (Thunderbird 91.5.0, Linux Mint 20.1)

Being no expert at all I hadn't realized how tricky this stuff is, sorry. Many thanks to @cleidigh for taking the burden! So far I'm happy with b14, just miss multiple parts selection. My test results:

Mail opened in a window or a tab or message pane shown:
Ctrl + P, Context Menu > Print (NG), AppMenu > Print (NG), Menu Bar > File > Print (NG), Mail Toolbar > Print NG, Message Header Toolbar > Print NG: PTNG-print;
Mail Toolbar > Print, Message Header Toolbar > Print: normal print
"Hide inline attachment list" and "Print just selection, when it exists" o.k. Printing multiple parts selected: first part printed only.

Still inconsistent, but no real problem: If the message pane is not shown and the mail is not opened, next time after the mail is printed or its print is previewed using PTNG   "Main Toolbar > Print" will also start PTNG-print instead of normal print until another mail has been printed or its print has been previewed (no matter if with normal print or PTNG-print).

(By the way, off-topic: Can anyone tell me how to hide the Message Header Toolbar without installing a special add-on?)

**Comment by cleidigh** [comment]
@querixg 

Thanks for the support and testing, yes there is quite a lot under the hood 
I am battling a print engine that does not like to be intercepted 

Two questions : 
- can you clarify the issues a bit more
- when  you mention multiple selections, I cannot see this was ever supported by ptng, does TB natively do this?
I ask because I am unable to try given my disability and the way my eye gaze keyboard works. Can you describe the key/mouse sequence?

Also posted b15 with full localization 
https://github.com/cleidigh/printing-tools-ng/blob/v2.0.7/xpi/printing-tools-ng-2.0.7-b15-tb.xpi
@cleidigh

**Comment by querixg** [comment]
2.0.7-b15 (Thunderbird 91.5.0, Linux Mint 20.1)

(Seemingly, I tend to put as much information as possible into a single sentence - German style maybe... Sorry.)

1.   Multiple selection print is possible with Firefox (and LibreOffice Writer etc.) and - Thunderbird normal print. Unfortunately, normal print with Format > Selection omits the header. Have a look:

Multiple selection (for example mark 'multiple' with the mouse and holding Ctrl mark 'print'):
![multiple selection](https://user-images.githubusercontent.com/87893156/155879293-641a7d74-1e03-4e4f-a8ad-2a51127c4768.png)

Normal print with Format > Selection:
![normal](https://user-images.githubusercontent.com/87893156/155879301-db3fc788-dff8-4091-bea3-d85fda600875.png)

PTNG-print with 'Print just selection':
![PTNG just selection](https://user-images.githubusercontent.com/87893156/155879310-3002beb3-8316-4720-99c4-1b156d72650b.png)

PTNG-print without 'Print just selection':
![PTNG](https://user-images.githubusercontent.com/87893156/155879313-95b0307d-97cd-4d87-85b9-73df47fa64ad.png)

2.   Inconsistency:
- Be sure you have chosen settings of PTNG that make PTNG-printout differ from normal printout.
- Have at least two mails in your mailing list.
- In View > Layout let message pane not be displayed.
- Don't open any mail.
(In the following steps it doesn't matter whether you print the mail or preview the printout only and cancel.)
- Mark a mail and choose (normal) Print: normal printout as expected.
- Choose Print (NG) for the same mail: PTNG-printout as expected.
- Choose (normal) Print for the same mail: PTNG-printout !
- Mark another mail and choose normal print or PTNG-print.
- Mark the first mail again and choose (normal) Print: normal printout.

3.   PTNG doesn't print eml-files.

**Comment by kiki-ja** [comment]
@querixg 

I have the following suggestion and request to @querixg.

**1.Multiple selection print**
Create a topic as a new issue.
The reason for this is that I think it is easier to separate reports from requests.

**2.Inconsistency**
The 2 pane view is understandable. (F8 = folder pane and thread pane)
What is the difference between Mark and Choose?
When PTNG is installed, normal printing cannot be selected.??
I don't know what the inconsistency is.
Please provide an easy to understand reproduction procedure.

**3.PTNG doesn't print eml-file**
You can already find the same below.
I have also listed them in the summary I wrote.

_"Extension does not work for e-mails which are saved to disk and for e-mails which are attachments to e-mails"_
https://github.com/cleidigh/printing-tools-ng/issues/5

I think it would be better to continue with that.
The reason is that duplicate issues are confusing and difficult to understand.

@kiki-ja

**Comment by cleidigh** [comment]
All
we desperately need to break out the final issues and make some decisions for release 

i am getting lost with everything mixed up 

@cleidigh

**Comment by tidebre67** [comment]
> All we desperately need to break out the final issues and make some decisions for release
> 
> i am getting lost with everything mixed up
> 
> @cleidigh

all,
yes, i losing the overlook as well. 

A lot of users waiting for an update of PTNG for TB 91.
As far as i can see the main functions are working pretty well and i guess
it was alreday a hell of work for Christopher to get so far.
Therefore i support that there should be a release as soon as possible.
Issues, if any, and additional features should be added  at a later stage.

**Comment by querixg** [comment]
@cleidigh
@tidebre67
> All we desperately need to break out the final issues and make some decisions for release

I think so too. As I said, I was happy with b14 and I am happy with b15. All further clarifications can wait.

**Comment by cleidigh** [comment]
@querixg 
Thanks for the detailed report, as @kiki-ja said let's break these out as separate issues with the multiple selection as a new feature request, which since tb supports this most likely we can.
I just don't think we should make that a release block.

On the eml issue highlight the Invocation methods.
Using File/Open/Saved Message (eml file)
works already, I cannot test opening a eml file from file explorer
if that's the issue posting debug output would help

@cleidigh

**Comment by querixg** [comment]
@cleidigh
@kiki-ja

#80 not yet closed, I take the opportunity for some final remarks so that I don't need to create new issues:

1. Multiple selection
By and then, when printing a mail I want to omit more than one part of the body. Since Thunderbird is able to print the header and is able to print multiple selected parts of the body I thought it would be a simple task to combine both in PTNG. If this is not the case then forget it! There is of course a simple alternative: print to PDF, copy to e.g. ODT, edit and print.

2. "Inconsistency"
Under very special circumstances that I described in detail, 'print' starts PTNG-print instead of normal print. These circumstances can be avoided easily, e.g. open the mail before printing it. I only reported it since it may be a hint to some bug in PTNG.

3. eml-files
I don't need PTNG for printing eml-files, but I tested it. If an eml-file is opened in Thunderbird (no matter whether you open it in Thunderbird by file > open saved message or choose Thunderbird as opening program in the file manager (Nemo in my case), normal print works. But contrary to remarks of Christopher, in my tests of b15 'Print NG' in all its variants didn't do anything. If I'm the only one who observes this you can forget it, too.

**Comment by reinerblock** [comment]
Sorry, can't currently participate because the SSD where my Windows operating system is located is completely defect since yesterday, and I'm using a notebook I got from my further education provider where I've my training to become a certified project manager.

The new SSD is announced to be delivered tomorrow. I really hope it is just the SSD and not a defect on the main board. I'm angry at myself because I wanted to make a backup the day before the SSD crashed, and now I am missing some general data and my profile data. Luckily, most of the data is on a different drive.
